snakemake_wrapper_utils/snakemake.py

```
@@ -45,3 +46,27 @@ def get_format(path):
+    """
+    Given the dict:
+    mapping = {"tsv": "/tmp/tmp98723489/out.tsv,}
+
+    Move file /tmp/tmp98723489/results/out.tsv to snakemake.output.tsv,
+    and redirect stdout and stderr to snakemake.log.
+    """
+
+    import contextlib
+
+    with open(snakemake.log[0], "a", buffering=1) as log:
+        with contextlib.redirect_stdout(log):
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)